Source/ARTMessage.m

```diff
@@ -45,3 +49,73 @@ - (NSInteger)messageSize {
 }
 
 @end
+
+@implementation ARTMessage (Decoding)
+
++ (instancetype)fromEncodedJsonObject:(NSDictionary *)json channelOptions:(ARTChannelOptions *)options error:(NSError **)error {
```

Comment:
How does this behave with VCDIFF-encoded messages? Given that the decoding of such a message relies on having a "previous" message available, I don't know how it's meant to work. The spec provides a recovery procedure in [RTL18](https://docs.ably.com/client-lib-development-guide/features/#RTL18), but it seems designed for the case where you have a realtime connection available to request the previous message. Do we need to address this in the spec?

Reply:
Probably externally saved messages shouldn't be VCDIFF-encoded?

Reply:
As far as I know, the Ably service will only ever send VCDiff-encoded messages on a Realtime connection. It is then the responsibility of the SDK to transparently apply that delta to the previous message and then deliver the result to the application. Therefore, I'm not sure that @lawrence-forooghian is describing a use-case that is possible, but I may be wrong.

Reply:
Does that mean we're happy to leave the behaviour undefined in this case?

Reply:
@QuintinWillison